pep-0701.txt

```diff
@@ -0,0 +1,367 @@
+PEP: 701
+Title: Syntactic formalization of f-strings
+Version: $Revision$
+Last-Modified: $Date$
+Author: Pablo Galindo <pablogsal@python.org>,
+        Batuhan Taskaya <batuhan@python.org>,
+        Lysandros Nikolaou <lisandrosnik@gmail.com>
+Discussions-To: discuss.python.org
+Type: Standards Track
+Status: Draft
+Content-Type: text/x-rst
+Python-Version: 3.12
+Created: 15-Nov-2022
+
+
+Abstract
+========
+
+This document proposes to lift some of the restrictions originally formulated in
+:pep:`498` and to provide a formalized grammar for f-strings that can be
+integrated into the parser directly. The proposed syntactic formalization of
+f-strings will have some small side-effects over how f-strings are parsed and
+interpreted, allowing for a considerable number of advantages for end users and
+library developers, while also improving dramatically the maintainance cost of
+the code dedicated to parse f-strings.
+
+
+Motivation
+==========
+
+When f-strings were originally introduced in :pep:`498` the specification was
+provided without providing a formal grammar for f-strings. Additionally, the
+specification contains several restrictions that are imposed so the parsing of
+f-strings could be implemented into CPython without modifying the existing
+lexer. These limitations have been recognized previously and previous attempts
+have been made to lift them in :pep:`536` but non of this work was ever implemented
+(see [2]_). Some of these limitations (collected originally by :pep:`536`) are:
+
+#. It is impossible to use the quote character delimiting the f-string
+   within the expression portion::
+
+    >>> f'Magic wand: { bag['wand'] }'
+                                 ^
+    SyntaxError: invalid syntax
+
+#. A previously considered way around it would lead to escape sequences
+   in executed code and is prohibited in f-strings::
+
+    >>> f'Magic wand { bag[\'wand\'] } string'
+    SyntaxError: f-string expression portion cannot include a backslash
+
+#. Comments are forbidden even in multi-line f-strings::
+
+    >>> f'''A complex trick: {
+    ... bag['bag']  # recursive bags!
+    ... }'''
+    SyntaxError: f-string expression part cannot include '#'
+
+#. Arbitrary nesting of expressions without expansion of escape sequences is
+   available in every single other language employing a string interpolation
+   method that uses expressions instead of just variable names. [6]_
```

Comment:
```suggestion
   method that uses expressions instead of just variable names, `per Wikipedia`_.
```

Make this an actual direct link, rather than an indirect footnote